### DISCUSSION TITLE
ML and 'AltAll' ancestral FASTA sequences from IQ-TREE .state files

### DISCUSSION THREAD

**Discussion by pdkiser** [iqtree/iqtree2#325] [Show and tell]
Hi Everyone,

First, thank you to the IQ-TREE development team for creating an easy-to-use implementation of ML ancestral sequence reconstruction. I want to let the IQ-TREE user community know about a small R package available in CRAN called 'maxaltall' that reads in an IQ-TREE .state file and outputs, in FASTA format, the ML sequence(s) as well as 'AltAll' sequence(s) that have uncertain sites (defined by the user) substituted with the next most likely state.

https://cran.r-project.org/web/packages/maxaltall/index.html

The 'AltAll' strategy is described here: https://doi.org/10.1093/molbev/msw223

Hopefully, the package will facilitate the selection of alternative sequence reconstructions that can be used to test the robustness of properties found for the ML ancestral sequence.

Yours,
Philip Kiser

**Comment by roblanf** [comment]
Love this! Wonderful to see things being built on top of IQ-TREE.

Mostly I'm posting to cross-reference this issue: https://github.com/iqtree/iqtree2/discussions/319, which talks about gaps in ancestral sequence reconstructions. I wonder if the posters on that thread might be interested in extending this package to do what they are interested in.
